testing/integration/k8s/otel_helm_test.go

```diff
@@ -165,32 +166,40 @@ func TestOtelKubeStackHelm(t *testing.T) {
 
 func k8sStepHelmDeployWithValueOptions(chartPath string, releaseName string, values values.Options) k8sTestStep {
 	return func(t *testing.T, ctx context.Context, kCtx k8sContext, namespace string) {
-		// Initialize a map to hold the parsed data
-		helmValues := make(map[string]any)
-
-		settings := cli.New()
-		settings.SetNamespace(namespace)
-		providers := getter.All(settings)
-		helmValues, err := values.MergeValues(providers)
-		if err != nil {
-			require.NoError(t, err, "failed to helm values")
-		}
+		helmValues := mergeValues(t, namespace, values)
 
 		k8sStepHelmDeploy(chartPath, releaseName, helmValues)(t, ctx, kCtx, namespace)
 	}
 }
 
+func mergeValues(t *testing.T, namespace string, values values.Options) map[string]any {
+	// Initialize a map to hold the parsed data
+	helmValues := make(map[string]any)
+
+	settings := cli.New()
+	settings.SetNamespace(namespace)
+	providers := getter.All(settings)
+	helmValues, err := values.MergeValues(providers)
+	if err != nil {
+		require.NoError(t, err, "failed to helm values")
+	}
+	return helmValues
+}
+
 // k8sStepCheckRunningPods checks the status of the agent inside the pods returned by the selector
 func k8sStepCheckRunningPods(podLabelSelector string, expectedPodNumber int, containerName string) k8sTestStep {
 	return func(t *testing.T, ctx context.Context, kCtx k8sContext, namespace string) {
-		require.Eventually(t, func() bool {
+		require.EventuallyWithTf(t, func(t *assert.CollectT) {
```

Comment:
Same point as above about the test fixture shadowing.

Reply:
done: https://github.com/elastic/elastic-agent/pull/11129/commits/38a5e6563b6c9c19fe4c2a52b7fd316505048ad2